packages/ui/src/lib/components/post.svelte

```diff
@@ -1,79 +1,89 @@
 <script lang="ts">
+	import Card from '$lib/components/grid-card.svelte'
 	import { formatDateAndTime } from '$lib/utils/format'
-
 	import type { Post } from '$lib/stores/post'
 
-	let cls: string | undefined = undefined
-	export { cls as class }
 	export let post: Post
 </script>
 
-<div class={`root ${cls}`}>
+<Card on:click>
 	<div class="content-wrapper">
+		<div class="imgs">
+			<!-- I HARD CODED SOME IMAGES TO STYLE THE SECTION AND LEFT THEM TO SHOW THE STRUCTURE IN USE -->
+			<!-- MORE THAN 3 IMAGES SHOULD HAVE A "PLUS" ICON OVER THE THIRD WITH COUNT OF EXTRA IMAGES -->
+
+			<div>
+				<img src="https://via.placeholder.com/300x500" alt="Placeholder for testing" />
+			</div>
+			<div>
+				<img src="https://via.placeholder.com/400x840" alt="Placeholder for testing" />
+			</div>
+			<div>
+				<img src="https://via.placeholder.com/100x75" alt="Placeholder for testing" />
+			</div>
+		</div>
+		<div class="post-content">{post.text}</div>
 		<div class="user-info">
 			<div class="faded">{formatDateAndTime(post.timestamp)}</div>
 		</div>
-		<div class="post-content">{post.text}</div>
 	</div>
-</div>
+</Card>
 
 <style lang="scss">
-	.root {
-		border-bottom: 1px solid var(--grey-200);
-		padding: var(--spacing-12);
+	.imgs {
 		display: flex;
-		gap: var(--spacing-12);
 		flex-direction: row;
-		break-inside: avoid-column;
+		gap: var(--spacing-6);
+		justify-content: flex-start;
+		align-items: center;
+		flex-wrap: nowrap;
 
-		@media (min-width: 640px) {
-			border: none;
-			outline-style: solid;
-			outline-width: 1px;
-			outline-color: var(--grey-200);
-			outline-offset: -0.5px;
+		div {
+			img {
+				max-height: 300px;
+			}
+			&:not(:only-child) img {
+				aspect-ratio: 1;
+				object-fit: cover;
+				width: 100%;
+				height: 100%;
+			}
 		}
 
-		@media (prefers-color-scheme: dark) {
-			border-bottom-color: var(--grey-500);
-			outline-color: var(--grey-500);
+		/* one item */
+		div:first-child:nth-last-child(1) {
+			width: 100%;
+		}
+
+		/* two items */
+		div:first-child:nth-last-child(2),
+		div:first-child:nth-last-child(2) ~ div {
+			width: 50%;
+		}
+
+		/* three items */
+		div:first-child:nth-last-child(3),
+		div:first-child:nth-last-child(3) ~ div {
+			width: 33.3333%;
```

Comment:
Pretty sure this calculation should not be here and instead we should use `flexBasis: 100%` :). https://youtu.be/rxnX1jdoI6c?t=398 and especially https://youtu.be/rxnX1jdoI6c?t=576